### ISSUE TITLE
[REQ] Add adjoints for `texture_sample`

### ISSUE THREAD

**Issue by eigenvivek** [NVIDIA/warp#1301]
I am trying to differentiable a dense voxelgrid w.r.t. the sampling position. In the example below, the gradient from warp is zero.

```python
import numpy as np
import warp as wp

wp.init()


@wp.kernel
def sample(tex: wp.Texture1D, pos: wp.array(dtype=float), out: wp.array(dtype=float)):
    tid = wp.tid()
    out[tid] = wp.texture_sample(tex, pos[tid], dtype=float)


data = np.arange(16, dtype=np.float32)
tex = wp.Texture1D(data, normalized_coords=False)
pos = wp.array([4.0], dtype=float, requires_grad=True)
out = wp.zeros(1, dtype=float, requires_grad=True)

tape = wp.Tape()
with tape:
    wp.launch(sample, dim=1, inputs=[tex, pos], outputs=[out])
out.grad = wp.ones(1, dtype=float)
tape.backward()

print("out:", out.numpy())            # [3.5]
print("pos.grad:", pos.grad.numpy())  # [0.] expected [1.]
```

The pytorch version returns the correct gradient (1).

```python
import torch
import torch.nn.functional as F

data = torch.arange(16, dtype=torch.float32).reshape(1, 1, 1, 1, 16)
pos = torch.tensor([4.0], requires_grad=True)

pos_norm = 2.0 * pos / 16.0 - 1.0
grid = torch.stack([pos_norm, torch.zeros_like(pos_norm), torch.zeros_like(pos_norm)], dim=-1)
grid = grid.reshape(1, 1, 1, 1, 3)  # [N, D, H, W, 3]
out = F.grid_sample(data, grid, mode="bilinear", align_corners=False)
out.backward()

print("out:", out.item())             # 3.5
print("pos.grad:", pos.grad.item())   # 1.0
```

Is texture sampling not currently differentiable or did I implement the gradient incorrectly? Also using the latest nightly.

**Comment by shi-eric** [comment]
A quick look at the source code and docs indicates that `texture_sample` is not currently differentiable. For example, the built-in https://nvidia.github.io/warp/language_reference/_generated/warp.texture_sample.html#warp._src.lang.texture_sample has no "Differentiable" tag (just "Kernel"), and the adjoints are currently empty stubs https://github.com/NVIDIA/warp/blob/ccb06a99d6db413ddbe64657cc8deeebdd58b1ab/warp/native/texture.h#L802-L846

We will consider the feature request to support the gradient propagation through the sampling positions.

@nvtw @nvlukasz for viz

**Comment by eigenvivek** [comment]
Ah I see, thank you for the pointer to the source! An adjoint for textures would be very helpful for a lot of optimization tasks. Will convert this to a request.
